### ISSUE TITLE
OBS becomes unresponsive when it can't reach the SRT Server

### ISSUE THREAD

**Issue by saltarob** [obsproject/obs-studio#7848]
### Operating System Info

macOS 12

### OBS Studio Version

28.1.2

### OBS Studio Log URL

https://obsproject.com/logs/g4eltzC1bo-WJHXm

### Expected Behavior

If OBS can't reach the SRT Server and and I stop the stream I expect the button to go back to the start stream position.

### Current Behavior

OBS reports that it connects and disconnects from the SRT Server. Basically it can't reach it. I am wondering why it reports a connection at all even if the server is down. If I stop the stream OBS gets caught in a stopping stream loop. It shows stopping stream forever and the stream bottom becomes unresponsive.

### Steps to Reproduce

1. SRT Server is down
2. Start the stream to the SRT server
3. Stop the stream to the SRT server
...

**Comment by saltarob** [comment]
Okay, but does the manual stop button not overwrite the auto reconnect attempts? That's what happens on Wirecast for example.

**Comment by tt2468** [comment]
This should be resolved in the new OBS 29 beta. @pkviet should be able to confirm.

**Comment by saltarob** [comment]
Great, thank you.

**Comment by pkviet** [comment]
Oh i thought the user was already testing in beta.
@saltarob test the beta. It should fix it. I wrote something to that end

**Comment by saltarob** [comment]
Still the same on OBS 29 beta1. Stream button gets stuck in the "connecting" state.

**Comment by saltarob** [comment]
May I ask something else? Is the SRT Stream monitored in the OBS stats? Will dropped frames in the SRT stream show up in the stats? Thank you.

**Comment by tt2468** [comment]
Can confirm, an SRT server denying new connections from OBS leaves the output in reconnecting mode and the UI prevents the user from cancelling.

**Comment by pkviet** [comment]
i'll push a fix tomorrow.

**Comment by pkviet** [comment]
Fix here https://github.com/obsproject/obs-studio/pull/7857 ; test the artefact once CI is done with building.

**Comment by pkviet** [comment]
test builds here: https://github.com/obsproject/obs-studio/actions/runs/3543698630

**Comment by saltarob** [comment]
Thank you. There is now this popup window which informs me that it can't connect to the server. Does this mean the automatic reconnect is bypassed now? Most of the time I am controlling OBS through the WebSockets. I think the popup Window can't be accessed through the WebSockets. Therefore the "Start Stream" button will be blocked. Can this popup Window be deactivated? I thought about stopping it manually with a second click on the stream button, which is showing "connecting". And as long as you don't cancel it manually it can retry automatically according the settings. That can be helpful if there is a short interruption from which it could recover automatically through reconnecting. And if you notice that the reconnection doesn't work you can stop it manually and check the settings etc.  Thank you.

**Comment by pkviet** [comment]
> Thank you. There is now this popup window which informs me that it can't connect to the server. Does this mean the automatic reconnect is bypassed now? Most of the time I am controlling OBS through the WebSockets. I think the popup Window can't be accessed through the WebSockets. Therefore the "Start Stream" button will be blocked. Can this popup Window be deactivated? I thought about stopping it manually with a second click on the stream button, which is showing "connecting". And as long as you don't cancel it manually it can retry automatically according the settings. That can be helpful if there is a short interruption from which it could recover automatically through reconnecting. And if you notice that the reconnection doesn't work you can stop it manually and check the settings etc. Thank you.

- automatic reconnect is stopped. This is in line with what happens with rtmp (just try with a faulty rtmp address): the same message is shown and auto-reconnect is also stopped.

- No the popup window either for rtmp or now for srt can not be deactivated. You have to click manually to dismiss the window.

- The popup won't show up in case of initial successful connection and later a network issue; the auto-reconnect will be triggered.

I understand that the popup window doesn't work with your current setup but there's no alternative, I'm afraid. I won't do anything different from rtmp. And your suggestion of stopping the stream manually won't work; the only option will be to fully exit obs (unless, if you were a dev, you  make a custom build to prevent the popups. If this is critical to you, consider hiring someone.)
Closing now as I consider this fixed.

**Comment by saltarob** [comment]
Okay I understand. Thank you so much. If it doesn't show up after the initial successful connection then it should be fine in most cases. And exit OBS would be the workaround in case the popup window is blocking the Start Stream button.

**Comment by tt2468** [comment]
Pop up dialogs have no effect on obs-wrbaocket functionality.

**Comment by saltarob** [comment]
Oh great. Just tested it with Touch Portal. OBS just opens another popup window. As often you click the "Start Streaming" button through WebSockets as many popup windows you get :-) Thank you so much.
